Sources/FoundationEssentials/Data/Data+Base64.swift

```diff
@@ -0,0 +1,374 @@
+            initializingWith: { //FIXME: should work with borrowed `bytes`
+                [bytes = copy bytes] in
+                try Data.base64DecodeBytes(bytes, &$0, options: options)
+            }
+        )
+    }
+
+    // MARK: - Create base64
+
+    /// Returns a Base-64 encoded string.
+    ///
+    /// - parameter options: The options to use for the encoding. Default value is `[]`.
+    /// - returns: The Base-64 encoded string.
+//    @available(macOS 11, *)
+    public func base64EncodedString(options: Base64EncodingOptions = []) -> String {
+        if self.isEmpty { return "" }
+
+        let capacity = Self.estimateBase64Size(length: self.count)
+        if #available(macOS 11, *) {
+            return self.withBufferView { inputBuffer in
+                String(utf8Capacity: capacity) {
+                    Self.base64EncodeBytes(inputBuffer, &$0, options: options)
+                }
+            }
+        }
+        else {
+          let ptr = UnsafeMutablePointer<UInt8>.allocate(capacity: capacity)
+          defer { ptr.deallocate() }
+          var outputBuffer = OutputBuffer(initializing: ptr, capacity: capacity)
+          self.withBufferView { inputBuffer in
+            Self.base64EncodeBytes(inputBuffer, &outputBuffer, options: options)
+          }
+          return outputBuffer.withBufferView {
+              String(decoding: $0, as: Unicode.UTF8.self)
+          }
+        }
+    }
+
+    /// Returns a Base-64 encoded `Data`.
+    ///
+    /// - parameter options: The options to use for the encoding. Default value is `[]`.
+    /// - returns: The Base-64 encoded data.
+    public func base64EncodedData(options: Base64EncodingOptions = []) -> Data {
+        let dataLength = self.count
+        if dataLength == 0 { return Data() }
+
+        return self.withBufferView { inputBuffer in
+            let capacity = Self.estimateBase64Size(length: dataLength)
+            return Data(capacity: capacity) {
+                Self.base64EncodeBytes(inputBuffer, &$0, options: options)
+            }
+        }
+    }
+
+    // MARK: - Internal Helpers
+
+    static func estimateBase64Size(length: Int) -> Int {
+        // Worst case allow for 64bytes + \r\n per line  48 input bytes => 66 output bytes
+        return ((length + 47) * 66) / 48
+    }
+
+    /**
+     Padding character used when the number of bytes to encode is not divisible by 3
+     */
+    static let base64Padding : UInt8 = 61 // =
+
+    /**
+     This method decodes Base64-encoded data.
+
+     If the input contains any bytes that are not valid Base64 characters,
+     this will throw a `Base64Error`.
+
+     - parameter bytes:      The Base64 bytes
+     - parameter output:     An OutputBuffer to be filled with decoded bytes
+     - parameter options:    Options for handling invalid input
+     - returns:              Whether decoding was successful
+     */
+    static func base64DecodeBytes(
+        _ bytes: borrowing BufferView<UInt8>,
```

Comment:
`bytes` is only iterated over; there doesn't seem to be any benefit to requiring a (contiguous) `BufferView`.

This could be generic. In fact, it could even be a single-pass `Sequence`.

Reply:
We do want to leave room for a byte-parallel implementation. We have not done that yet, but it is eminently doable, and more so with contiguous storage. Currently we are replicating the existing API (and ABI) from the existing version of Foundation. We may add more API at a later time.